endpoints/openrtb2/auction_test.go

```diff
@@ -1222,6 +1212,54 @@ func TestCCPAInvalid(t *testing.T) {
 	assert.Empty(t, req.Regs.Ext, "Invalid Consent Removed From Request")
 }
 
+func TestValidateSourceTID(t *testing.T) {
+	cfg := &config.Configuration{
+		AutoGenSourceTID: true,
+	}
```

Comment:
I have added a very minimal test based on the setup we have today. Ideally we'd wanna add a table driven test with multiple test cases but validateRequest is a big function and we'd have to refactor all the existing one-off validateRequest tests to convert into table driven tests. That will make this PR very crowded and so it'd be better for that to be its own separate PR.